src/m3ninx/search/proptest/query_gen.go

```diff
@@ -54,7 +55,8 @@ func genTermQuery(docs []doc.Document) gopter.Gen {
 	}
 }
 
-func genRegexpQuery(docs []doc.Document) gopter.Gen {
+// GenRegexpQuery generates a regexp query.
+func GenRegexpQuery(docs []doc.Document) gopter.Gen {
```

Comment:
for the failure case i was describing, you probably need to change this generator to flip a biased coin and choose between this implementation and mapping the output of a TermQuery to a RegexpQuery

Reply:
Same comment

Reply:
Hm are you sure this is capable of generating the same thing as a term query?

Reply:
oh no I was just saying that the interface doesn't allow this to occur